service/history/archival_queue_task_executor.go

```diff
@@ -343,7 +345,10 @@ func (e *archivalQueueTaskExecutor) loadAndVersionCheckMutableState(
 		task,
 	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf(
+			"mutable state last write version does not match task version: %v",
+			err,
```

Comment:
the error handling logic can't handle this wrapped error. 
https://github.com/temporalio/temporal/blob/master/service/history/queues/executable.go#L282

Reply:
Fixed, thank you!